src/components/organism/SwapTokens/index.tsx

```diff
@@ -477,9 +477,18 @@ const SwapTokens = () => {
       }
     }
   };
-
+  const getSwapTokenB = () => {
+    const poolLiquidTokens: any = [nativeToken]
+      .concat(poolsTokenMetadata)
+      ?.filter(
+        (item: any) =>
+          item.tokenId !== selectedTokens.tokenA?.tokenId && item.tokenId !== selectedTokens.tokenB?.tokenId
+      );
+    return poolLiquidTokens;
+  };
   const fillTokenPairsAndOpenModal = (tokenInputSelected: TokenSelection) => {
-    getPoolTokenPairs().then((res: any) => setAvailablePoolTokens(res));
+    if (tokenInputSelected === "tokenA") getSwapTokenA().then((res: any) => setAvailablePoolTokens(res));
+    if (tokenInputSelected === "tokenB") setAvailablePoolTokens(getSwapTokenB());
```

Comment:
for both lines here u should setState in the function itself. 
Eg: instead of setAvailablePoolTokens(getSwapTokenB()); u can setAvailablePoolTokens inside of getSwapTokenB